### PR TITLE
Add a push-on-to action for split view controller, where the detail i…

### PR DESCRIPTION
**Pull request by benronicus**
Adds a split view action to push on a new view controller to the split view's detail stack. I wasn't sure if it would be appropriate to create a navigation controller for the detail if missing within the action, so routing will fail if the navigation controller isn't there.

### CODE REVIEW

**Comment by ekazaev** [comment]
> I wasn't sure if it would be appropriate to create a navigation controller for the detail if missing within the action, so routing will fail if the navigation controller isn't there.

Thank you for the PR. I thought about it, and i do not think it is right to do it. It would be some "magic" that happens underneath the navigation. I think it should not happen and the developer should be aware about what is he doing, and be able to fix it and have a full control.

File: RouteComposer/Classes/Actions/UISplitViewController+Action.swift [hunk 1]
```diff
@@ -83,4 +87,39 @@ public struct SplitViewControllerActions {
...
+      /// Constructor
+      init() {
+      }
+
+      public func perform(embedding viewController: UIViewController, in childViewControllers: inout [UIViewController]) throws {
+        guard !childViewControllers.isEmpty else {
+          throw RoutingError.compositionFailed(.init("Master view controller is not set in " +
+            "UISplitViewController to push on a detail view controller \(viewController)."))
+        }
+        childViewControllers.append(viewController)
```
Comment (ekazaev) [inline RouteComposer/Classes/Actions/UISplitViewController+Action.swift]: @benronicus I have a doubt here tbh. 
This action requires a `UINavigationController` to be the last in stack to be successfully performed. See `perform(with ,on, animated, completion:)`,
But this method that is used for the instant build of the whole `UISplitViewController` doesn't have this check. So it will create a situation when using the same action may behave differently. I think the `viewControllers as? UINavigationController` should be added along with the check that there are not more than 2 `UINavigationController`s in the `childViewControllers` stack. But i am not hugely familiar with the split controller and its sideeffects. So i am curious about your thoughts about it. I would prefer action to not allow to shoot in to the leg, basically.

Comment (benronicus) [reply RouteComposer/Classes/Actions/UISplitViewController+Action.swift]: Apologies for the slow response, just seeing this. I've been using the action without issue but  I'll give it some more thought and come back with a new PR if I see improvements to be made. Can you clariy what you mean by "this method that is used for the instant build"?

Comment (ekazaev) [reply RouteComposer/Classes/Actions/UISplitViewController+Action.swift]: @benronicus sorry. English is not my native language. So it means that this method will be used by the router to build the whole view controller stack from scratch. Imagine the situation that you need to build navigation controller with 2 view controllers inside. Router will call this method twice saying i have this view controller I don’t know how to build it into the array of view controllers that i will later put into your navigation controller. 

So basically the nature of your actions than it expects split view controller to be build out of the navigation controllers. And it should not let other view controllers in the stack and throw the exception. Similar how the method below does it

Comment (ekazaev) [reply RouteComposer/Classes/Actions/UISplitViewController+Action.swift]: @benconstantine39 @benronicus I actually added this change myself after i merged your PR. So you can see what i meant in the latest version
